Update ESLint

AbdullahAli opened the pull request:
A few updates to make the code nicer

- [x] Use the most up-to-date ESLint and 2018 syntax (async/await, spread, etc)
- [x] Force object spread `...` (🙌🏽) over `Object.assign` (🤢)

AbdullahAli commented on app/services/queue.js:
```diff
@@ -504,39 +503,38 @@ class QueueApp extends EventEmitter {
-							});
-							sendToSentry(email);
+		}).then(logRejections)
+			.catch((errorObj) => {
```
Not sure why github makes it look like that (it isn't)

keirog commented on app/controllers/send.controller.js:
```diff
@@ -28,7 +28,7 @@ const scsCategoryMap = new Map([
 	const categorization = userInfo.categorization.find(c => c.domain === domain);
 	//TODO: get rid of lodash
-	return Object.assign({ product: userInfo.product }, _.pick(categorization, 'binding', 'returnPath', 'category'));
+	return { product: userInfo.product, ..._.pick(categorization, 'binding', 'returnPath', 'category') };
```
Yessss

keirog commented on app/util/partially-apply-logger.js:
```diff
@@ -1,5 +1,5 @@
 module.exports = ({ rawLogger, data }) => {
 	return (logLevel, extraData) => {
-		rawLogger[logLevel](Object.assign({}, data, extraData));
+		rawLogger[logLevel]({ ...data, ... extraData });
```
~`... extraData`~ `...extraData`

keirog commented on test/middleware/emailContent.test.js:
```diff
@@ -57,7 +57,7 @@ describe('EmailContent middleware', () => {
 	it('throws error if fetched email content is marked inactive', done => {
 		nextSpy = sandbox.spy();
-		const inactiveEmailContent = Object.assign({}, emailContent, { active: false });
+		const inactiveEmailContent = { ... emailContent, active: false };
```
~`... emailContent`~ `...emailContent`

keirog commented on test/models/userModel.test.js:
```diff
@@ -34,7 +34,7 @@ describe('User Model', () => {
 
 	it('updates an existing user', function* () {
-		const updatedDetails = Object.assign({}, userData, { description: 'updated user' });
+		const updatedDetails = { ... userData, description: 'updated user' };
```
~`... userData~ `...userData`

keirog commented on test/util/substitutionData.test.js:
```diff
@@ -10,14 +10,14 @@ const userSubData = [{
 describe('The substitutionData util', () => {
 	describe('merge', () => {
-		const recipients = [Object.assign({}, userInfo)];
+		const recipients = [{... userInfo }];
```
~`... userInfo`~ `...userInfo`

keirog commented on test/util/substitutionData.test.js:
```diff
@@ -10,14 +10,14 @@ const userSubData = [{
 
 		it('does not do anything if no user substitution data', () => {
-			const recipients = [Object.assign({}, userInfo)];
+			const recipients = [{... userInfo }];
```
~`... userInfo`~ `...userInfo`

AbdullahAli commented:
Ah, good spot @keirog 

I have added eslint rules to ensure these don't get missed next time (re: `...` spacing)

Fixed: https://github.com/Financial-Times/email-send-api/pull/262/commits/f790167b18dba86d93eb43a95dd8f8fba8580b1a

AbdullahAli commented:
It would be good to extend established eslint styleguides - like [Airbnb Styleguide](https://github.com/airbnb/javascript)
